### ISSUE TITLE
Callback in async_connection::read() is bad

### ISSUE THREAD

**Issue by dmytrobogovych** [cpp-netlib/cpp-netlib#768]
Greetings!

async_connection.hpp (with http support), line 395. Callback param list is bad in version 0.12 - this can be download from site. 
Seems call to this->wrap_read_handler() is missed. 
Ubuntu 14.04, gcc 4.8.4, cpp-netlib 0.12 (download from site).

**Comment by dmytrobogovych** [comment]
To be precise - it is problem of 0.12 distribution files only.
Git files are ok.

**Comment by deanberris** [comment]
Thanks for the report. Yes, but can you try 0.13-devel to see if this is already fixed there too?

**Comment by dmytrobogovych** [comment]
Yes, it is fine in 0.13. Master is fine too (I switched to it after all experiments).
